platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbDataSource.java

```diff
@@ -369,7 +371,15 @@ public MerkleDbDataSource(
                 loadedDataCallback = (path, dataLocation, keyValueData) -> {
                     // read key from keyValueData, as we are in isLongKeyMode mode then
                     // the key is a single long
-                    final long key = keyValueData.getLong(0);
+                    final long key;
+                    // This is ugly, but hopefully we'll be able to drop this code altogether soon
```

Comment:
Could you please create a follow-up ticket about dropping this code with a description of what exactly it should be replaced with?

Reply:
I checked this code again, and I feel it's broken today and needs to be reworked or dropped. In particular, `keyValueData` is actually data item data, which in case of leaf records is a serialized path/key/value. The very first long is a path, not a key. The key is the second long (given that keys are longs in this mode). However, it relies particular binary format of serialized virtual leaf records. The format can be changed in the future, and is actually changed in this PR for PBJ. That's why I think it should be changed or dropped